### PR TITLE
[FIX][12.0] postgres 14 compatibility for anyarray

### PR DESCRIPTION
**Pull request by sergiocorato**
Postgres 14.x is incompatible with `anyarray`.

This PR make use of `anycompatiblearray` for postgres >= 14 and of 'anyarray` for prior versions.

### CODE REVIEW

**Comment by NL66278** [comment]
@sergiocorato Unfortunately this fails on postgres versions before 14.0. So you must either:
1. Check for the postgres version and use either anyarray or anycompatible array depending on version;
2. Or try one of the options and if it fails try the other.

**Comment by NL66278** [comment]
@sergiocorato I created a PR on your branch: https://github.com/sergiocorato/account-financial-reporting/pull/1

**Review by NL66278 — APPROVED** [review]
:+1: LGTM And used in production environment

**Comment by sergiocorato** [comment]
@OCA/accounting-maintainers other opinions?

**Review by thomaspaulb — APPROVED** [review]

**Review by Bilbonet — APPROVED** [review]
LGTM 👍

**Comment by Bilbonet** [comment]
Hi @sergiocorato 
It would be nice to merge this if it was possible.
Thanks in advance.

**Comment by sergiocorato** [comment]
@Bilbonet I opened #1050 as this PR cannot be merged without having updated the repo to copier template.

**Comment by sergiocorato** [comment]
@thomaspaulb @NL66278 this needs #1050 to be merged

**Comment by sergiocorato** [comment]
Superseded by #1056

**Pull request closed without merging**
